lib/howl/buffer.moon

```diff
@@ -277,6 +278,19 @@ class Buffer extends PropertyObject
       b_end_pos = @byte_offset end_pos
       @_buffer\ensure_styled_to pos: b_end_pos
 
+  mode_at: (pos) =>
+    b_pos = @byte_offset pos
+    mode_name = @_buffer.styling\get_mode_name_at b_pos
+    -- Returns @mode if there's no marker or the requested mode doesn't exist.
+    mode_name and mode.by_name(mode_name) or @mode
+
+  config_at: (pos) =>
+    mode_at = @mode_at pos
+    return @config if mode_at == @mode
+    new_config = config.local_proxy!
```

Comment:
So this is something I'm not sure about. As @shalabhc mentioned in an earlier comment this loses any buffer specific configuration when in a sub mode. Is this reasonable, or would one expect something else? Like I said I'm not sure - perhaps this is the way it should work since buffer specific setting were probably set considering the parent mode in the first place, and it might be preferable to use the mode's config directly. To prevent any confusion one could in that case return a read-only proxy, or otherwise it would possibly to return a new read-only chained config respecting both local and sub mode settings. That we could address post-merge however, but we need to figure out what the expected behaviour would be.